### PR TITLE
[AIRFLOW-4108] Load autocomplete data asynchronously

### PR DESCRIPTION
**Pull request by mik-laj**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

This PR introduces changes:
- loading data into typeahead in an asynchronous way
- add sorting 
- fixes searching by the owner field

CC: @astahlman 

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: airflow/www/templates/airflow/dags.html [hunk 1]
```diff
@@ -215,6 +215,7 @@ <h2>DAGs</h2>
   <script>
 
       const DAGS_INDEX = "{{ url_for('Airflow.index') }}";
+      const DAGS_AUTOCOMPLETE = "{{ url_for('Airflow.dags_autocomplete') }}";
```
Comment (mik-laj) [inline airflow/www/templates/airflow/dags.html:218]: I know about the existence of an PR, but this change has not been accepted yet, so I follow the old way.
https://github.com/apache/airflow/pull/4787/files
Depending on which change will be accepted first, the relevant PR will update.

File: airflow/www/views.py [hunk 2]
```diff
@@ -282,9 +282,46 @@ def get_int_arg(value, default=0):
...
+    @provide_session
+    def dags_autocomplete(self, session):
+        arg_search_query = request.args.get('search', None)
+
+        if not arg_search_query:
+            return wwwutils.json_response({
+                'items': []
+            })
+
+        max_items_count = int(request.args.get('limit', 8))
```
Comment (mik-laj) [inline airflow/www/views.py:298]: This is not needed at the moment, but I wanted to make the server code independent of the client's configuration. Typeahead by default displays only 8 elements.

**Review by ashb — COMMENTED** [review]
Could you add tests that cover:

- searching by owner
- that dags the user doesn't have access to don't show up?

File: airflow/www/views.py [hunk 3]
```diff
@@ -282,9 +282,46 @@ def get_int_arg(value, default=0):
...
+        filter_dag_ids = appbuilder.sm.get_accessible_dag_ids()
+
+        dags_query = session.query(DagModel.owners).filter(
+            ~DagModel.is_subdag, DagModel.is_active
```
Comment (ashb) [inline airflow/www/views.py:302]: Did the previous view show paused dags? (i.e. is this a change in behaviour?)

Comment (mik-laj) [reply airflow/www/views.py:302]: Yes. It always redirected to a view that shows all DAGs. https://github.com/apache/airflow/pull/4929/files/04953cffe59151d07a5ff44c448a57584c4247bd#diff-f38558559ea1b4c30ddf132b7f223cf9L271

**Pull request closed without merging**